### PR TITLE
Reduce coupling with 3rd-party libraries

### PR DESCRIPTION
**Pull request by simonbrandhof**
Plugin artifact is 7.4MB. It's quite big. Most of the size is for the required XML libs. Guava could be dropped but it's still needed by sslr-squid-bridge. 
Nevertheless this pull request does some clean-up and reduces the artifact size to.... only 6.6MB (baby step !). It also fixes some metadata in pom.xml files.

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [SchemaResolver.java#L239](https://github.com/SonarSource/sonar-xml/blob/48b7daf5bd268f615059638fd3b07295295d4e0b/sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/schemas/SchemaResolver.java#L239): Disambiguate this call by either casting as "Object" or "Object[]". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3878)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [XmlFile.java#L156](https://github.com/SonarSource/sonar-xml/blob/99f4c7639a6d8840c7c9df118d142913425681c8/sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/XmlFile.java#L156): Replace this fully qualified name with "Files" [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1942)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by dbmeneses** [comment]
Added some more changes, size is now at **2.4MB**.

**Comment by m-g-sonar** [comment]
LGTM @simonbrandhof & @dbmeneses for now. I'm gonna continue your effort and reduce dependency on `sslr-squid-bridge` as well

File: sonar-xml-plugin/src/test/java/org/sonar/plugins/xml/rules/XmlRulesDefinitionTest.java [hunk 1]
```diff
@@ -43,6 +44,11 @@ public void test() {
     assertThat(alertUseRule).isNotNull();
     assertThat(alertUseRule.name()).isEqualTo("Source code should be indented consistently");
 
+    assertThat(Stream.of(repository.rule("XPathCheck"), repository.rule("XmlSchemaCheck")))
+      .isNotEmpty()
+      .allMatch(rule -> rule != null)
+      .allMatch(rule -> rule.template());
```
Comment (pynicolas) [inline sonar-xml-plugin/src/test/java/org/sonar/plugins/xml/rules/XmlRulesDefinitionTest.java]: Thats doesn't check that only these 2 rules are declared as rule templates.

**Pull request merged**